File: mobile/lib/logic/format.dart
import 'package:intl/intl.dart';

final _price = NumberFormat.currency(locale: 'tr_TR', symbol: '₺', decimalDigits: 0);
final _num = NumberFormat.decimalPattern('tr_TR');
final _score = NumberFormat('0.0', 'tr_TR');

String formatPrice(int value) => value == 0 ? 'Fiyat yok' : _price.format(value);
String formatNumber(num value) => _num.format(value);
String formatScore(num value) => _score.format(value);

String formatCount(int value) {
  if (value >= 1000) {
    return '${_score.format(value / 1000)} B';
  }
  return _num.format(value);
}

String boolLabel(bool value) => value ? 'Var' : 'Yok';

String formatCapacity(int? capacity, [String? note]) {
  if (capacity == null) return note ?? 'Üretici belirtmedi';
  return '${formatNumber(capacity)} mAh';
}

String formatHours(int hours, [String? note]) {
  if (hours == 0) return note ?? 'Belirtilmedi';
  return note == null ? '$hours saat' : '$hours saat · $note';
}

File: src/lib/format.ts
const trNumber = new Intl.NumberFormat("tr-TR");
const trCurrency = new Intl.NumberFormat("tr-TR", {
  style: "currency",
  currency: "TRY",
  maximumFractionDigits: 0,
});

export function formatNumber(value: number) {
  return trNumber.format(value);
}

export function formatPrice(value: number) {
  if (!value) return "Fiyat yok";
  return trCurrency.format(value);
}

export function formatScore(value: number) {
  if (!Number.isFinite(value)) return "—";
  return value.toLocaleString("tr-TR", {
    minimumFractionDigits: 1,
    maximumFractionDigits: 1,
  });
}

export function formatCount(value: number) {
  if (value >= 1000) {
    return `${(value / 1000).toLocaleString("tr-TR", { maximumFractionDigits: 1 })} B`;
  }
  return trNumber.format(value);
}

export function boolLabel(value: boolean) {
  return value ? "Var" : "Yok";
}

export function formatCapacity(capacity: number | null, note?: string) {
  if (capacity == null) {
    return note ?? "Üretici belirtmedi";
  }
  return `${formatNumber(capacity)} mAh`;
}

export function formatHours(hours: number, note?: string) {
  if (!hours) return note ?? "Belirtilmedi";
  return note ? `${hours} saat · ${note}` : `${hours} saat`;
}
